src/App.jsx

```diff
@@ -14,45 +17,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ABC = 'abc';
+const SORT_FIELD_LENGTH = 'length';
+const RESET_FIELD = '';
+
+export const App = () => {
+  let visibleGoods = [...goodsFromServer];
+  const [reverse, setReverse] = useState(false);
+  const [sortField, setSortField] = useState('');
+  const hasResetButton = reverse || sortField;
+
+  const handleSort = field => () => setSortField(field);
+  const handleReverse = reversed => () => setReverse(!reversed);
+  const handleReset = (reverseReset, sortReset) => () => {
+    setReverse(reverseReset);
+    setSortField(sortReset);
+  };
+
+  if (reverse) {
+    if (sortField) {
+      visibleGoods.sort((good1, good2) => {
+        switch (sortField) {
+          case SORT_FIELD_ABC:
+            return good2.localeCompare(good1);
+
+          case SORT_FIELD_LENGTH:
+            return good1.length - good2.length;
+
+          default:
+            return 0;
+        }
+      });
+    }
+
+    if (!sortField) {
+      visibleGoods.reverse();
+    }
+  } else {
+    if (sortField) {
```

Comment:
Don't need to duplicate sort functionality 
You need to execute sort only once, after that check isReverse field, if it true, than just return reversed list